src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -1221,7 +1221,7 @@ class StubGenerator: public StubCodeGenerator {
       }
       __ addptr(qword_count, 4);
       __ BIND(L_end);
-      if (UseAVX >= 2) {
+      if (UseAVX >= 1) {
         // clean upper bits of YMM registers
         __ vpxor(xmm0, xmm0);
```

Comment:
Should we use `vzeroupper` to clear the upper half of `ymm`s?

Reply:
No. That goes against [JDK-8078113](https://bugs.openjdk.java.net/browse/JDK-8078113).

Reply:
Oh, in that case, I think we should not bother zeroing out those anyway since `vpxor` here would not serve the purpose that zeroing out upper bits is for (SSE - AVX transition).

Reply:
Oh! I think I understand the concern here. Current vpxor is called with VEX.256 (vector_len == 1), which does  [MAXVL-1:256] ← 0. vzeroupper, on the other hand, does [MAXVL-1:128] ← 0. This looks like a pre-existing bug that is not directly related to this patch, as it affects UseAVX = 2 code too. Let me figure this out separately.

Reply:
That `vpxor` is not even usable on AVX=1, produces SIGILL (MacOS builds in GHA, and some of my platforms). So I pushed the temporary workaround that both works on AVX=1 and matches vzeroupper better. Is that what you were looking for?

Reply:
Hi,

The legacy SSE instructions require the bits [MAXVL - 1 : 128] of register XMM0 - 15 to remain intact. As a result, the processors employ a state tracking mechanism with 2 possible circumstances, clean and dirty upper bits. When the upper bits are dirty, SSE instructions need to either save the content of all those bits on entry of the SSE code block or do a blend on every SSE instruction, which is costly. On the contrary, when the upper bits are in the clean state, SSE code can execute freely without additional operations.

The processors switch from the clean state to the dirty state when an instruction that touches the upper bits is executed (e.g a VEX256 instruction); and revert back to the clean state when some special instructions such as `vzeroupper` and `vzeroall` are executed. This state is global, that is a VEX256 instruction on `xmm0` will affect SSE instructions on every other register, as a result, `vpxor` is not an instruction possible to instruct the processor to transit to the clean state.
This is why we need to do a `vzeroupper` at the end of a compiled method whenever a vector instruction with larger-than-128-bit width is present.

More precise details can be found [here](https://www.naic.edu/~phil/software/intel/11MC12_Avoiding_2BAVX-SSE_2BTransition_2BPenalties_2Brh_2Bfinal.pdf).

Thanks.

Reply:
There is a `vzeroupper` at the end of arraycopy stub, so AVX<->SSE state transitions are handled here. Before [JDK-8078113](https://bugs.openjdk.java.net/browse/JDK-8078113) we had `vzeroupper` in these places, but apparently it clobbers too much of the YMM state.

But I suspect this issue only affects the "real" intrinsics (like in `String`), not the arraycopy stubs. There are no SSE instructions after leaving the AVX copy code up to arraycopy stub, where we meet `vzeroupper`. So this `vpxor`-ing might just be redundant.

Reply:
Understood, thank you very much for clearing that out. It seems weird that we try to use `vzeroupper` in the middle of the compiled code considering hotspot never emits SSE instructions on AVX machines.

Reply:
See https://bugs.openjdk.java.net/browse/JDK-8279676 for more work about this.